tests/testthat/test-add_group.R

```diff
@@ -0,0 +1,13 @@
+test_that("add_group adds a group as expected", {
+  expect_equal(
+    epinowcast:::add_group(data.frame(x = 1)),
+    data.table::data.table(x = 1, .group = 1)
+  )
+})
+
+test_that("add_group doesn't add a group when one is present", {
+  expect_equal(
+    epinowcast:::add_group(data.frame(x = 1, .group = 1)),
+    data.frame(x = 1, .group = 1)
+  )
+})
```

Comment:
should test this with a different pre-existing group, since this is the group that would be created anyway.